### DISCUSSION TITLE
Speaker diarization with speaker names

### DISCUSSION THREAD

**Discussion by nftaper** [deepgram/community#475] [General help] [answered]
Hi everyone! I'm new here and was wondering if it is possible to perform speaker diarization with the actual speaker names instead of "speaker 1", "speaker 2".

Any help would be much appreciated!

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by jkroll-deepgram** [comment, marked as answer]
Hi @nftaper, welcome to Deepgram! Our diarization only provides numeric speaker labels. You'll need to do your own additional processing to associate speaker numbers with names. Depending on your use case, speaker order can help - i.e. if you know that a certain person would speak first on a call, you can infer that Speaker 0 is that person.

**Reply by nftaper** [reply]
Thanks for the response! Will try automating this too

**Reply by akuten1298** [reply]
@nftaper Were you able to solve this? would love to get some insights

**Comment by geopopos** [comment]
I ended up creating a quick solution for this.

I use the utterances and diarization options for my transcription.

from there I create a script with the first 50 utterances of my call


```
import json
from datetime import timedelta

def seconds_to_timestamp(seconds):
    # Convert seconds to a timedelta object
    td = timedelta(seconds=seconds)
    
    # Format the timedelta object as a string
    # The total_seconds() method is used to handle cases where the input seconds
    # exceed the range of a single day, ensuring the hours part is correctly calculated.
    hours, remainder = divmod(td.total_seconds(), 3600)
    minutes, seconds = divmod(remainder, 60)
    milliseconds = (seconds - int(seconds)) * 1000
    
    # Format the string with leading zeros where necessary
    formatted_time = "{:02d}:{:02d}:{:06.3f}".format(int(hours), int(minutes), seconds)
    
    return formatted_time
   

utterances = YOUR RESPONSE FROM THE DEEPGRAM API

transcript = ""
for index, utterance in enumerate(utterances):
    print(utterance)
    start_timestamp = seconds_to_timestamp(utterance["start"])
    end_timestamp = seconds_to_timestamp(utterance["end"])
    formatted_string = f"""{index + 1}
{start_timestamp} --> {end_timestamp}
Speaker {utterance["speaker"]}: {utterance["transcript"]}\n\n"""
    transcript += formatted_string
    
```

From there I use the openai API to determine my speakers
USER MESSAGE:
THE TRANSCRIPT YOU JUST CREATED
SYSTEM MESSAGE
```
You are an expert at determining speakers names based on transcripts.

Your job is to determine the names of the speakers in the transcript.

Use common knowledge like if someone is saying "Hey George" that likely means the other speakers name is George, not the speaker saying hi.

This is spoken language.

Your only output should be a json object with the index as the key and the value being the speaker name you decided on.

EX:
{
"0": "Speakers Name",
"1":  "Speakers Name"
}
```

Response_Format: json_object
Temperature: 0.2


after that I have a dictionary that I can use with the speaker key value pair in the utterances dictionary to create a transcript that uses speaker name instead of numbers

**Reply by geopopos** [reply]
Also make sure to use gpt-4 or higher here. gpt-3.5 consistently reversed the names.
